### PR TITLE
Add scripts, docs for migrating `message_data` code

### PR DESCRIPTION
**Pull request by khaeru**
These describe/implement the process used for #88.

This PR can be updated as #88 (or #86, or another PR) is completed and merged.

## How to review

Read the updated documentation.

## PR checklist

- [x] Complete and merge #88; rebase this branch.
- [x] Add a reference to the info added to the description of #86 by @glatterf42.
- [x] Continuous integration checks all ✅
- [x] ~Add or expand tests;~ coverage checks both ✅
- [x] Add, expand, or update documentation.
- [x] Update doc/whatsnew.

### CODE REVIEW

File: doc/migrate.rst [hunk 1]
```diff
@@ -38,11 +41,197 @@ Use the :program:`mix-models` command-line interface (CLI)
...
+      $ ./2-migrate.sh
+
+   Among other things, this script:
+
+   - Clones :mod:`message_ix_models`.
+   - Copies the Git LFS objects from the :mod:`.message_data` clone in step (4), so that they are available for use in the target location.
+   - Uses :program:`git-filter-repo` (`docs <https://htmlpreview.github.io/?https://github.com/newren/git-filter-repo/blob/docs/html/git-filter-repo.html>`_) to:
+
+     - Move code.
+       For example, all commits pertaining to a file like :file:`message_data/model/foo/bar.py` are preserved, except they they now appear to describe changes to :file:`message_ix_models/model/foo/bar.py`.
```
Comment (glatterf42) [inline doc/migrate.rst:117]:
```suggestion
       For example, all commits pertaining to a file like :file:`message_data/model/foo/bar.py` are preserved, except they now appear to describe changes to :file:`message_ix_models/model/foo/bar.py`.
```
A double 'they' has sneaked up here.

**Comment by khaeru** [comment]
For reference, here are the duplicates.txt and (manually edited) rebase-todo.txt produced by 2-migrate.sh while applying this process for #88:
- [duplicates.txt](https://github.com/iiasa/message-ix-models/files/10822934/duplicates.txt)
- [rebase.txt](https://github.com/iiasa/message-ix-models/files/10822935/rebase.txt)

For example:
- `Add water infrasructure` [sp] as a commit message appears in duplicates.txt —the development history of the source repo/branch has resulted in duplicates of this commit.
- So, in editing rebase-todo.txt to rebase.txt, I removed one of the occurrences of this commit in the list of commands.

**Comment by khaeru** [comment]
@glatterf42 I've now completed the text and TODO list above, so this is ready to be reviewed and, if approved, merged.

**Review by glatterf42 — APPROVED** [review]
Very clear and concise, lgtm :)

**Pull request merged**